DSharpPlus/Entities/AutoModeration/DiscordAutoModerationRule.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using DSharpPlus.Enums;
+using DSharpPlus.Net.Models;
+
+using Newtonsoft.Json;
+
+namespace DSharpPlus.Entities;
+
+/// <summary>
+/// Represents a Discord auto-moderation rule.
+/// </summary>
+public class DiscordAutoModerationRule : SnowflakeObject
+{
+    /// <summary>
+    /// Gets the id of the guild which the rule is in.
+    /// </summary>
+    [JsonProperty("guild_id")]
+    public ulong GuildId { get; internal set; }
```

Comment:
Prefer exposing entities (DiscordGuild) where possible. Use the provided client from `SnowflakeObject`